code/processes/tailreader.q

```diff
@@ -1,40 +1,71 @@
-\d .tr 
+\d .tr
 partitiontype:@[value;`partitiontype;`date];                               /-set type of partition (defaults to `date)
 gmttime:@[value;`gmttime;1b];                                              /-define whether the process is on gmttime or not
-determinepartition:{{@[value;`.tr.currentpartition;                        /-function to determine the partition value
-    (`date^partitiontype)$(.z.D,.z.d)gmttime]}
-  }; 
-getpartition:@[value;`getpartition;determinepartition[]];                  /-check if partition value exists and if not generate one
+determinepartition:{@[value;`.tr.currentpartition;                         /-function to determine the partition value
+  (`date^partitiontype)$(.z.D,.z.d)gmttime]
+  };
+getpartition:@[value;`getpartition;determinepartition[]];
 segmentid: "J"$.proc.params[`segid]
-taildir:hsym `$getenv`KDBTAIL						   /-load in taildir env variables
-.tr.currentpartition:.tr.getpartition;					   /-obtain  partition value
+taildir:hsym `$getenv`KDBTAIL                                              /-load in taildir env variables
+currentpartition:getpartition;                                             /-obtain  partition value
 basedir:(raze/)1_string[.tr.taildir],"/tailer",string .tr.segmentid,"/"       /-define associated tailer base directory
-wdbdir:`$ basedir,string currentpartition                                      /-define IDB direction
+wdbdir:`$ basedir,string currentpartition                                  /-define IDB direction
+tailertypes:`$"tailer_",last "_" vs string .proc.proctype                  /-define tailer to make connection to 
+.servers.CONNECTIONS:(distinct .servers.CONNECTIONS,.tr.tailertypes) except ` 
+.servers.startup[];
+
+
+\d .ds
+
+getaccess:{[] `location`table xkey update location:.proc.procname,proctype:.proc.proctype from .ds.access};
+
+// function to update the access table in the gateway. Takes the gateway handle as argument
+updategw:{[h]
+
+    newtab:getaccess[];
+    neg[h](`.ds.updateaccess;newtab);
+
+    };
 
 \d .
 endofday:{[pt]
   /- end of day function that will be triggered by EOD Sorter once IDB is copied to HDB
   /-  updates partition and loads in next days partition
   .lg.o[`eod;"End of day message received - ",spt:string pt];
-    currentpartition::pt+1;
-    reload[];
+  .tr.currentpartition:pt+1;
+  reload[];
   }
 
 reload:{
   /- function to define the access table and IDB dir and then reload both tables
   /- reload is triggered by tailer after savedown occurs
-  basedir::(raze/)1_string[.tr.taildir],"/tailer",string .tr.segmentid,"/";
-  wdbdir::`$ basedir,string currentpartition;
-  accesstabdir::`$ basedir,"access";
-  
- @[.Q.l ;wdbdir;{.lg.e[`load;"Failed to load intradayDB with error: ",x]}];
- @[.Q.l ;accesstabdir;{.lg.e[`load;"Failed to load tailer accesstable with error: ",x]}];
-  load hsym `$basedir,"sym"
+  .tr.basedir:(raze/)1_string[.tr.taildir],"/tailer",string .tr.segmentid,"/";
```

Comment:
This looks a bit messy. Often nice to build filepaths using `sv` https://code.kx.com/q/ref/sv/#filepath-components

(similar in next few lines)

Reply:
This looks to be old code which hasn't been updated on this branch.  I believe the plan will be to use the format shown in the PR #538 for the final merge.

Reply:
hmm, it shows in the diff, are you sure it hasn't been updated on this branch?

Reply:
Sorry, I meant it hasn't been updated on this branch since an early stage - we had been working on this code elsewhere. I'm going to update this script to a final version here so that we can be sure to have the latest work all in one place.